test/e2e/specs/editor/various/list-view.spec.js

```diff
@@ -0,0 +1,345 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const openListView = async ( { page, pressKeyWithModifier } ) => {
+	// Open List View with keyboard.
+	await pressKeyWithModifier( 'access', 'o' );
+	// Wait until the current item is selected to ensure UI is ready.
+	await page
+		.getByRole( 'gridcell', {
+			name: /link/i,
+			selected: true,
+		} )
+		.waitFor();
+};
+
+test.describe( 'List View', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'allows a user to drag a block to a new sibling position', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a couple of blocks of different types.
+		await editor.insertBlock( { name: 'core/heading' } );
+		await editor.insertBlock( { name: 'core/image' } );
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		// Open the List View panel.
+		await openListView( pageUtils );
+
+		// Ensure the setup is correct before dragging.
+		await expect
+			.poll( editor.getBlocks )
+			.toMatchObject( [
+				{ name: 'core/heading' },
+				{ name: 'core/image' },
+				{ name: 'core/paragraph' },
+			] );
+
+		// Drag the paragraph above the heading.
+		const paragraphBlockItem = page.getByRole( 'gridcell', {
```

Comment:
I actually did that, but then had second thoughts and [removed it](https://github.com/WordPress/gutenberg/pull/47919/commits/23d9daff7e74f9a5fcd40613bbe1a937d8b4c95f). 😅 The reason behind removing it was because it adds an unnecessary selector which can potentially make the test more fragile. I agree, though, that it would increase the readability of the code - I'll bring it back.

Reply:
> it adds an unnecessary selector which can potentially make the test more fragile.

I don't think it would make the test more fragile though. On the contrary, I think it will help us catch some unintentional bugs. For instance, imagine there is a gridcell role with the same name that exists in some other part of the page, and the original one for some reason disappears. It's very unlikely though 😅.

Reply:
I thought about it and assumed the accessible name of the parent (`Block navigation structure`) would change sooner than another `gridcell` with a matching name would appear in the DOM. 😄 That difference is likely negligible, though, so increasing the readability is a good tradeoff. 👍

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/47919/commits/80a52315cee116a768951862197552aa3809f1c5.